### DISCUSSION TITLE
How many single file binaries for windows, mac, and linux?

### DISCUSSION THREAD

**Discussion by zapta** [pyinstaller/pyinstaller#7457] [Help]
I have an open source python app with 20+ source files so decided to generate also single file binaries for windows, mac, and linux (I understand that I need to build on each target type). My question is, how many binaries I need to build to cover most of the 'market' of these three systems. 

E.g. one for windows (doesn't matter if intel or amd), two for mac (e.g. one for intel and one for Arm) and a few for ubuntu (e.g. one for ubuntu intel/amd, and one for another distro?).

Any suggestion of what binaries to generate?

**Comment by bwoodsend** [comment]
That's completely up to you and your users. Personally, I feel that anyone one Linux will be tech savvy enough to run Python code from source provided that it's [properly packaged](https://packaging.python.org/en/latest/) and PyInstaller makes so little sense on Linux anyway that I'd just stick to Windows and macOS.

**Comment by zapta** [comment]
What about macosx. Does pyinstaller require separate binaries for Arm and
Intel architectures?

**Reply by bwoodsend** [reply]
Technically Arm can run Itel binaries under emulation but yes, 2 for mac.

**Reply by zapta** [reply]
thanks @bwoodsend.  Is this plan reasonable?

Windows:  1. buill on intel architecture.
Mac:  1. built on intel architecture, 2. built on arm architecture
Linux:  1. build on intel architecture.

Rest will use the python source code.

**Reply by bwoodsend** [reply]
Sounds sensible. Just make sure that you use an old Linux (e.g. Ubuntu 18) when you build the Linux one - otherwise it's unlikely to run on most Linux machines.

**Comment by zapta** [comment]
Thanks
